chain/client/src/sync.rs

```diff
@@ -201,7 +189,7 @@ impl HeaderSync {
             *sync_status =
                 SyncStatus::HeaderSync { current_height: header_head.height, highest_height };
             self.syncing_peer = None;
-            if let Some(peer) = highest_height_peer(highest_height_peers) {
+            if let Some(peer) = highest_height_peers.choose(&mut thread_rng()).cloned() {
```

Comment:
nit: do you need cloned here?

Reply:
yeah I think here it is needed actually, because of the `self.syncing_peer =` part, so the struct has to be cloned somewhere eventually